include/xtensor/xutils.hpp

```diff
@@ -464,17 +464,18 @@ namespace xt
         template <class R>
         struct sequence_forwarder<R, R>
         {
-            static inline const R& run(const R& r)
+            template <class A>
+            static inline auto run(A&& r)
             {
                 return r;
             }
         };
     }
 
     template <class R, class A>
-    inline auto forward_sequence(const A& s)
+    inline auto forward_sequence(A&& s)
     {
-        return detail::sequence_forwarder<R, A>::run(s);
+        return detail::sequence_forwarder<R, A>::run(std::forward<A>(s));
```

Comment:
hum with `R = vector` and `A = vector &`, we might never match `<R, R>`... We can't decay because of static arrays.